controllers/iamauthpolicy_controller.go

```diff
@@ -0,0 +1,68 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
+
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type IAMAuthPolicyController struct {
+	log    gwlog.Logger
+	client client.Client
+}
+
+func RegisterIAMAuthPolicyController(log gwlog.Logger, mgr ctrl.Manager) error {
+	controller := &IAMAuthPolicyController{
+		log:    log,
+		client: mgr.GetClient(),
+	}
+	err := ctrl.NewControllerManagedBy(mgr).
+		For(&anv1alpha1.IAMAuthPolicy{}).
```

Comment:
You could add the `pkg_builder.WithPredicates(predicate.GenerationChangedPredicate{})` in the later PRs

Reply:
[GenerationChangedPredicate description](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/predicate#GenerationChangedPredicate) is a bit vague to me. Especially "Caveats" section. Why do we use this?